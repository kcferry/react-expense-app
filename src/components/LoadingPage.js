import React from 'react';
import loader from '../../public/images/loader.gif'

const LoadingPage = () => (
    <div className='loaderGif'>
        <img className="loader__image" src={loader} alt="Page loading gif"/>
    </div>
)

export default LoadingPage
