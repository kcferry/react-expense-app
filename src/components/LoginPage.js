import React from 'react'
import { connect } from 'react-redux'
import { startLoginGoogle, startLoginGithub } from '../actions/auth'

export const LoginPage = ({startLoginGoogle, startLoginGithub}) => (
    <div className="box-layout">
        <div className="box-layout__box">
            <h1 className="box-layout__title">Expensify</h1>
            <p>It's time to get your expenses under control</p>
            <div className='box-layout__buttons'>
                <button onClick={startLoginGoogle} className="button">Login with Google</button>
                <button onClick={startLoginGithub} className="button">Login with GitHub</button>
            </div>
        </div>
    </div>
)

const mapDispatchToProps = (dispatch) => ({
     startLoginGoogle: () => dispatch(startLoginGoogle()),
     startLoginGithub: () => dispatch(startLoginGithub())
})

export default connect(undefined, mapDispatchToProps)(LoginPage)


