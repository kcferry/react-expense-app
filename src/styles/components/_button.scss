.button {
    background: $light-blue;
    font-weight: 300;
    display: inline-block ;
    font-size: $font-size-large;
    padding: $s-size $l-size;
    color: white;
    line-height: 1;
    border: none;
    text-decoration: none;
}

.button-link {
    background: none;
}

.button:focus{  
    outline: none;
 }