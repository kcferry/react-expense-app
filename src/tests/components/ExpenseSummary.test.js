import React from 'react'
import { shallow } from 'enzyme'
import { ExpenseSummary } from '../../components/ExpenseSummary'

test('Should render the summary with 1 expense', () => {
    const wrapper = shallow(<ExpenseSummary expenseCount={1} expenseTotal={110} />)
    expect(wrapper).toMatchSnapshot()
});

test('Should render the summary with multiple expenses', () => {
    const wrapper = shallow(<ExpenseSummary expenseCount={12} expenseTotal={1112387640} />)
    expect(wrapper).toMatchSnapshot()
})