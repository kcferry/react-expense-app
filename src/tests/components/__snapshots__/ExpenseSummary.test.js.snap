// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the summary with 1 expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
       Viewing 
      <span>
         
        1
         
      </span>
       
      expense
       totalling 
      <span>
        $1.10
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`Should render the summary with multiple expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
       Viewing 
      <span>
         
        12
         
      </span>
       
      expenses
       totalling 
      <span>
        $11,123,876.40
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
